devicetypes/smartthings/thirdreality-waterleak-sensor.src/thirdreality-waterleak-sensor.groovy

```diff
@@ -0,0 +1,111 @@
+/**
+ *  ThirdReality WaterLeak Sensor
+ *
+ *  Copyright 2021 THIRDREALITY
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ */
+metadata {
+    definition (name: "ThirdReality WaterLeak Sensor", namespace: "smartthings", author: "THIRDREALITY", cstHandler: true) {
+        capability "Battery"
+        capability "Switch"
+        capability "Water Sensor"
+        capability "Refresh"
+        capability "Configuration"
+
+        fingerprint profileId: "0104", deviceId: "0402", inClusters: "0000,0001,0500", outClusters: "0006,0019", manufacturer:"Third Reality, Inc", model:"3RWS18BZ", deviceJoinName: "Water Leak Sensor"		//ThirdReality WaterLeak Sensor
+    }
+
+    simulator {
+        // When simulating, define status and reply messages here
+    }
+
+    tiles {		//Seems no use
+        // define your main and details tiles here
+        main("water")
+        details(["water", "battery", "switch"])
+    }
+}
+
+// parse events into attributes
+def parse(String description) {
+    log.trace "[parse] Parsing '${description}'"
+    def resMap = [:]
+
+    log.debug "[parse] descMap: $descMap"
+    if (description?.startsWith("zone status")) {
+        resMap = createEvent(name: "water", value: zigbee.parseZoneStatus(description).isAlarm1Set() ? "wet" : "dry")
+        sendEvent(name: "switch", value: zigbee.parseZoneStatus(description).isAlarm1Set() ? "on" : "off")
+    } else if (description?.startsWith("on/off")) {
+        def event = zigbee.getEvent(description)
+        sendEvent(event)
+    } else if (description?.startsWith("read attr")) {
+    	def descMap = zigbee.parseDescriptionAsMap(description)
+        if (descMap?.cluster == "0001" && descMap?.attrId == "0021") {
```

Comment:
A small style preference, but could you change these hex strings to either our enum types (e.g. `zigbee.POWER_CONFIGURATION_CLUSTER` `zigbee.ON_OFF_CLUSTER`) or define "constants" and compare to `descMap?.clusterInt` and `descMap?.attrInt`?